### ISSUE TITLE
[MASSEMBLY-530] Allow configuration of encoding 

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-assembly-plugin#696]
**[Max Schaefer](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=awpxw)** opened **[MASSEMBLY-530](https://issues.apache.org/jira/browse/MASSEMBLY-530?redirect=false)** and commented

Zips, that encode the file names with e.g. UTF-8 are not properly unpacked because the system default encoding is assumed.
I tracked down that PlexusIoZipFileResourceCollection initialises a ZipFile instance passing in just the file. However, a second parameter of ZipFile allows to control the encoding. I would like to be able to specify that encoding when unpacking a dependencySet.

---

**Affects:** 2.2

1 votes, 5 watchers

**Comment by jira-importer** [comment]
**[Dmitry Pelevin](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dpelevin)** commented

1) You can use plexus-archiver 2.4.4 (it contain fix for LEF flag).
2) I just fin out that it is possibe to set encoding for maven-archiver-plugin. Below is an example of configuration. The only issue is that you can set encoding ONLY for zip archiver - other will fail as they do not have setEncoding() method. So you need to split packaging to different formats into several \<execution>s.

```xml
<plugin>
  <artifactId>maven-assembly-plugin</artifactId>
  <version>2.4</version>
  <configuration>
    <descriptors>
      <descriptor>package-sources.xml</descriptor>
      <descriptor>package.xml</descriptor>
    </descriptors>
    <archiverConfig>
      <encoding>CP866</encoding>
    </archiverConfig>
  </configuration>
  <executions>
    <execution>
      <id>make-assembly</id>
      <phase>package</phase>
      <goals>
        <goal>single</goal>
      </goals>
    </execution>
  </executions>
  <dependencies>
    <dependency>
      <groupId>org.codehaus.plexus</groupId>
      <artifactId>plexus-archiver</artifactId>
      <version>2.4.4</version>
      <exclusions>
        <exclusion>
          <groupId>org.codehaus.plexus</groupId>
          <artifactId>plexus-component-api</artifactId>
        </exclusion>
      </exclusions>
    </dependency>
  </dependencies>
</plugin>
```

I think this ticket can ba closed as the only issue with encoding configuration is support from plexus-archiver. Zip already have encoding support and there is my pull request for TAR format: https://github.com/sonatype/plexus-archiver/pull/11.

**Comment by jira-importer** [comment]
**[Kristian Rosenvold](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=krosenvold)** commented

Archvier 2.5 defaults to UTF-8 and suports the LEF flag, which should be good enough. THis is fixed for 2.5
